lib/shopify_cli/environment.rb

```diff
@@ -105,23 +105,18 @@ def self.infer_spin?(env_variables: ENV)
     end
 
     def self.spin_url(env_variables: ENV)
-      if infer_spin?(env_variables: env_variables)
-        # TODO: Remove version check and delete spin-legacy branch
-        #  once spin2 becomes the installed "spin" binary by default
-        spin_version = %x(spin version 2> /dev/null).strip
-        if spin_version.start_with?("spin-")
-          # spin2
-          raise ShopifyCLI:: Abort, "SPIN_INSTANCE must be specified" unless ENV.key?("SPIN_INSTANCE")
-          %x(spin show -o fqdn 2> /dev/null).strip
-        else
-          # spin-legacy
-          %x(spin info fqdn 2> /dev/null).strip
-        end
+      if ENV.key?("SPIN_INSTANCE")
+        %x(spin show -o fqdn 2> /dev/null).strip
+      elsif infer_spin?(env_variables: env_variables)
+        instances = JSON.parse(%x(exe/spin list --json --output created,fqdn))
+
+        last_created_instance = instances.sort do |instance_a, instance_b|
+          DateTime.parse(instance_b["created"]) <=> DateTime.parse(instance_a["created"]) 
+        end.first
```

Comment:
I believe this would do the trick and is a tiny bit easier to read:

```suggestion
        last_created_instance = instances.min_by do |instance|
          DateTime.parse(instance["created"])
        end
```